cli/pkg/edgeclient/client.go

```diff
@@ -96,13 +102,370 @@ func (c *Client) AddNewModelWithImage(
+		}
+	}
+
+	md, err := modelregistry.FromMetadataValueMap(v.GetCustomProperties())
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to ensure version is in model registry: %w", err)
+	}
+	return v, md, nil
+}
+
+func (c *Client) ensureArtifactIsInModelRegistry(modelVersionID, artifactName, description, modelFormatName, modelFormatVersion string) error {
+	artifact, err := c.modelRegistryClient.FindModelVersionArtifact(modelVersionID, artifactName)
+	if err != nil {
+		if errors.Is(err, modelregistry.ErrFindArtifact) {
+			// TODO: Figure out what to do with the URI
+			_, err := c.modelRegistryClient.CreateModelArtifact(
+				modelVersionID, artifactName, description, "", modelFormatName, modelFormatVersion,
+			)
+			if err != nil {
+				return fmt.Errorf("failed to ensure artifact is in model registry: %w", err)
+			}
+		} else {
+			return fmt.Errorf("failed to ensure artifact is in model registry: %w", err)
+		}
+	} else if artifact == nil {
+		// Should never happen as we get an error if the artifact is not found, but just in case
+		return fmt.Errorf("failed to ensure artifact is in model registry: artifact not found")
+	}
+
+	return nil
 }
 
-func shortHash(s string) string {
-	hasher := sha256.New()
-	hasher.Write([]byte(s))
-	hash := hex.EncodeToString(hasher.Sum(nil))
-	return hash[:4]
+// CreatePipelineRun creates a tekton PipelineRun to build a model container image from a model version artifact.
+func (c *Client) CreatePipelineRun(
+	modelName string,
+	modelVersion string,
+	namespace string,
+	kubeConfig string,
+	parameters map[string]interface{},
+) (*PipelineRun, error) {
+	var s3SecretName string
+	var testDataConfigMapName string
+	var addDirWorkspace bool
+
+	if s3s, ok := parameters["s3SecretName"]; !ok {
+		return nil, fmt.Errorf("s3SecretName pipeline parameter is required")
+	} else if s3s, ok := s3s.(string); !ok {
+		return nil, fmt.Errorf("s3SecretName pipeline parameter must be a string")
+	} else {
+		s3SecretName = s3s
+	}
+
+	if tdc, ok := parameters["testDataConfigMapName"]; !ok {
+		return nil, fmt.Errorf("testDataConfigMapName pipeline parameter is required")
+	} else if tdc, ok := tdc.(string); !ok {
+		return nil, fmt.Errorf("testDataConfigMapName pipeline parameter must be a string")
+	} else {
+		testDataConfigMapName = tdc
+	}
+
+	if adw, ok := parameters["addDirWorkspace"]; !ok {
+		return nil, fmt.Errorf("addDirWorkspace pipeline parameter is required")
+	} else if adw, ok := adw.(string); !ok {
+		return nil, fmt.Errorf("addDirWorkspace pipeline parameter must be a boolean")
+	} else {
+		addDirWorkspace = adw == "true"
+	}
+
+	params, err := toTektonParams(modelName, modelVersion, parameters)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert parameters to tekton params: %w", err)
+	}
+
+	pipelineRun := &tektonv1.PipelineRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    namespace,
+			GenerateName: "aiedge-e2e-",
```

Comment:
It might make sense for this to include the model name also. We've started doing that with our example PipelineRuns, e.g.: https://github.com/opendatahub-io/ai-edge/blob/main/pipelines/tekton/aiedge-e2e/aiedge-e2e.tensorflow-housing.pipelinerun.yaml#L7

Reply:
fixed